### PR TITLE
MAINT: small simplification of meson.build following best practices

### PR DESCRIPTION
**Pull request by dnicolodi**
Just a couple of things I observed having a quick look at `meson.build`.

### CODE REVIEW

**Comment by WillAyd** [comment]
@lithomas1

File: pandas/_libs/window/meson.build [hunk 1]
```diff
@@ -3,7 +3,6 @@ py.extension_module(
     ['aggregations.pyx'],
     cython_args: ['-X always_allow_keywords=true'],
     include_directories: [inc_np, inc_pd],
-    dependencies: [py_dep],
```
Comment (lithomas1) [inline pandas/_libs/window/meson.build:6]: Can you do the same for ``pandas/_libs/meson.build`` and ``pandas/_libs/tslibs/meson.build``?

Comment (dnicolodi) [reply pandas/_libs/window/meson.build:6]: Unless I misunderstanding the request, these other `meson.build` files do not contain the problematic construct.

**Comment by WillAyd** [comment]
Another cleanup I noticed in the context of the optimization discussion we had is that every extension right now picks up CFLAGS and CPPFLAGS, yet we only have one extension module that actually uses c++ (window/aggregations.pyx)

Would be nice if we didn't mix those up (doesn't need to be done in this PR)

**Comment by lithomas1** [comment]
thanks @dnicolodi

**Comment by phofl** [comment]
This broke the windows builds

**Comment by dnicolodi** [comment]
> This broke the windows builds

Didn't CI pass on these changes? How does the build break?

**Comment by dnicolodi** [comment]
I see. I didn't notice the problem before this PR was merged. Sorry for the trouble.

The issue is this:
```
..\..\meson.build:2:0: ERROR: Command `C:\hostedtoolcache\windows\Python\3.9.13\x64\python3.EXE generate_version.py --print` failed with status 1.
```
This is the command that generates the version string in the `project()` call. I changed that from using `python` to `python3`. I don't understand why that fails though. It seems that `python3` is found.

**Comment by dnicolodi** [comment]
If you need a quick resolution, can you please revert just the last commit in this PR, and not the whole thing?

**Comment by dnicolodi** [comment]
Curiously, the `python-dev (windows-latest)` job passes. While the other Windows build fail. I wonder if it is a build environment issue. It would be interesting to see the content of the meson log in `D:\a\pandas\pandas\build\cp311\meson-logs\meson-log.txt`

**Comment by lithomas1** [comment]
Ah sorry, didn't realize it was the Windows not the npdev that was failing.

**Pull request merged**